homeassistant/components/http/decorators.py

```diff
@@ -12,20 +13,61 @@
 
 _HomeAssistantViewT = TypeVar("_HomeAssistantViewT", bound=HomeAssistantView)
 _P = ParamSpec("_P")
+_FuncType = Callable[
+    Concatenate[_HomeAssistantViewT, Request, _P], Coroutine[Any, Any, Response]
+]
 
 
+@overload
 def require_admin(
-    func: Callable[Concatenate[_HomeAssistantViewT, Request, _P], Awaitable[Response]]
-) -> Callable[Concatenate[_HomeAssistantViewT, Request, _P], Awaitable[Response]]:
+    _func: None = None,
+    *,
+    error: Unauthorized | None = None,
+) -> Callable[[_FuncType[_HomeAssistantViewT, _P]], _FuncType[_HomeAssistantViewT, _P]]:
+    ...
+
+
+@overload
+def require_admin(
+    _func: _FuncType[_HomeAssistantViewT, _P],
+) -> _FuncType[_HomeAssistantViewT, _P]:
+    ...
+
+
+def require_admin(
+    _func: _FuncType[_HomeAssistantViewT, _P] | None = None,
```

Comment:
@cdce8p Awesome!! Thanks a lot for fixing it :rocket: 

My question is that I don't understand/know why we need to specify arguments for the alias `_FuncType`.
And why only a few and not all? `Request` is missing.

Reply:
> My question is that I don't understand/know why we need to specify arguments for the alias `_FuncType`. And why only a few and not all? `Request` is missing.

Both `_HomeAssistantViewT` and `_P` are generics within the context of the generic type alias `_FuncType`. Whenever you use `_FuncType` (or any other generic type for that mater), you should specify the types for the generic types. Otherwise they are assumed to be `Any`. I.e. if `_FuncType` should continue to be a generic, you need to use `_FuncType[_HomeAssistantViewT, _P]` again.